server/services/timelineService.test.ts

```diff
@@ -0,0 +1,212 @@
+import type { TimelineResponse } from 'educationAndWorkPlanApiClient'
+import type { Timeline } from 'viewModels'
+import PrisonService from './prisonService'
+import TimelineService from './timelineService'
+import { EducationAndWorkPlanClient, HmppsAuthClient } from '../data'
+import { toTimeline } from '../data/mappers/timelineMapper'
+
+jest.mock('../data/mappers/timelineMapper')
+
+describe('timelineService', () => {
+  const mockedTimelineMapper = toTimeline as jest.MockedFunction<typeof toTimeline>
+
+  const hmppsAuthClient = {
+    getSystemClientToken: jest.fn(),
+  }
+
+  const educationAndWorkPlanClient = {
+    getTimeline: jest.fn(),
+  }
+
+  const prisonService = {
+    getPrisonByPrisonId: jest.fn(),
+  }
+
+  const timelineService = new TimelineService(
+    hmppsAuthClient as unknown as HmppsAuthClient,
+    educationAndWorkPlanClient as unknown as EducationAndWorkPlanClient,
+    prisonService as unknown as PrisonService,
+  )
+
+  beforeEach(() => {
+    jest.resetAllMocks()
+  })
+
+  const prisonNumber = 'A1234BC'
+  const username = 'a-dps-user'
+  const userToken = 'a-user-token'
+  const systemToken = 'a-system-token'
+
+  describe('getTimeline', () => {
+    describe('lookup prison names', () => {
```

Comment:
This test class didnt exist previously. I've started it by writing tests specifically for the prison name lookup stuff that it does, but I have not wriiten tests for the rest of the functionality in this class. We will need to do that 👍